source/grid/grid_generator.cc

```diff
@@ -5845,6 +5845,96 @@ namespace GridGenerator
+
+    std::array<Triangulation<dim, spacedim>, dim> partially_replicated_trias;
+    const Triangulation<dim, spacedim> *          tria_to_replicate = &input;
+    for (unsigned int d = 0; d < dim; ++d)
+      {
+        std::vector<Point<spacedim>> input_vertices;
+        std::vector<CellData<dim>>   input_cell_data;
+        SubCellData                  input_subcell_data;
+        std::tie(input_vertices, input_cell_data, input_subcell_data) =
+          GridTools::get_coarse_mesh_description(*tria_to_replicate);
+        std::vector<Point<spacedim>> output_vertices     = input_vertices;
+        std::vector<CellData<dim>>   output_cell_data    = input_cell_data;
+        SubCellData                  output_subcell_data = input_subcell_data;
+
+        for (unsigned int k = 1; k < extents[d]; ++k)
+          {
+            const std::size_t vertex_offset = k * input_vertices.size();
+            for (const Point<spacedim> &point : input_vertices)
+              output_vertices.push_back(point + double(k) * offsets[d]);
+            for (const CellData<dim> &cell_data : input_cell_data)
+              {
+                output_cell_data.push_back(cell_data);
+                for (unsigned int &vertex : output_cell_data.back().vertices)
+                  vertex += vertex_offset;
+                if (dim > 1)
+                  for (const CellData<1> &boundary_line :
+                       input_subcell_data.boundary_lines)
+                    {
+                      output_subcell_data.boundary_lines.push_back(
+                        boundary_line);
+                      for (unsigned int &vertex :
+                           output_subcell_data.boundary_lines.back().vertices)
+                        vertex += vertex_offset;
+                    }
+                if (dim > 2)
+                  for (const CellData<2> &boundary_quad :
+                       input_subcell_data.boundary_quads)
+                    {
+                      output_subcell_data.boundary_quads.push_back(
+                        boundary_quad);
+                      for (unsigned int &vertex :
+                           output_subcell_data.boundary_quads.back().vertices)
+                        vertex += vertex_offset;
+                    }
+              }
+          }
+
+        // check all vertices: since the grid is coarse most will be on the
+        // boundary anyway
+        std::vector<unsigned int> boundary_vertices;
+        GridTools::delete_duplicated_vertices(
+          output_vertices,
+          output_cell_data,
+          output_subcell_data,
+          boundary_vertices,
+          1e-6 * input.begin_active()->diameter());
+        // delete_duplicated_vertices also deletes any unused vertices
+        // deal with any reordering issues created by delete_duplicated_vertices
+        GridReordering<dim>::reorder_cells(output_cell_data, true);
+        partially_replicated_trias[d].create_triangulation(output_vertices,
```

Comment:
Is there any reason that you save all the intermediate triangulations instead of just keeping one temporary object?

Reply:
No - this is a leftover from development. I force pushed a new version without this.